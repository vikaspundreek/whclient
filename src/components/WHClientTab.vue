<template>
  <div
    class="wh-client__tab nav-link"
    :class="isActive ? 'active' : ''"
    @click="onClick"
  >
    {{ tabLabel }}
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from "vue-property-decorator";

@Component
export default class WHClientTab extends Vue {
  @Prop() private tabLabel!: string;
  @Prop() private isActive!: boolean;
  onClick(): void {
    !this.isActive && this.$emit("onTabClick");
  }
}
</script>
