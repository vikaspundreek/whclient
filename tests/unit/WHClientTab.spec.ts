import { shallowMount } from "@vue/test-utils";
import WHClientTab from "@/components/WHClientTab.vue";

describe("WHClientTab.vue", () => {
  const tabLabel = "Inventory";
  it("renders tab with correct title and active true state", () => {
    const wrapper = shallowMount(WHClientTab, {
      propsData: { tabLabel, isActive: true },
    });
    expect(wrapper).toMatchSnapshot();
  });
  it("renders tab with correct title and active false state", () => {
    const wrapper = shallowMount(WHClientTab, {
      propsData: { tabLabel, isActive: false },
    });
    expect(wrapper).toMatchSnapshot();
  });
  it("should emit onTabClick with correct param when clicked on non active tab", () => {
    const wrapper = shallowMount(WHClientTab, {
      propsData: { tabLabel, isActive: false },
    });
    wrapper.find(".wh-client__tab").trigger("click");
    expect(wrapper.emitted("onTabClick")).not.toBe(undefined);
  });
  it("should not emit onTabClick when clicked on active tab", () => {
    const wrapper = shallowMount(WHClientTab, {
      propsData: { tabLabel, isActive: true },
    });
    wrapper.find(".wh-client__tab").trigger("click");
    expect(wrapper.emitted("onTabClick")).toBe(undefined);
  });
});
