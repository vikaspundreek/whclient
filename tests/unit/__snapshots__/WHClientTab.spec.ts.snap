// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WHClientTab.vue renders tab with correct title and active false state 1`] = `<div class="wh-client__tab nav-link">Inventory</div>`;

exports[`WHClientTab.vue renders tab with correct title and active true state 1`] = `
<div class="wh-client__tab nav-link active">
  Inventory
</div>
`;
